content/open-source/wolfi/wolfi-with-dockerfiles.md

```diff
@@ -0,0 +1,221 @@
+
+```shell
+mv Dockerfile _DockerfileBkp
+```
+
+Then, create a new Dockerfile:
+
+```shell
+nano Dockerfile
+```
+
+Copy the following content to it:
+
+```
+FROM cgr.dev/chainguard/wolfi-base
+
+RUN adduser -D nonroot
+WORKDIR /app
+
+RUN apk add python-3.11 py3.11-pip && \
+	chown -R nonroot.nonroot /app/
+
+USER nonroot
+COPY requirements.txt inky.png inky.py /app/
+RUN  pip3 install -r requirements.txt --user
+
+ENTRYPOINT [ "python", "/app/inky.py" ]
+```
+
+Save the file when you're done. In the next step, we'll build and run the image with `docker`.
+
+
+## Step 3: Building and Running the Image
+
+With the Dockerfile ready, you can now build your application runtime. If you're on macOS, make sure Docker is running.
+
+Build your image with:
+
+```shell
+docker build . -t inky-demo
+```
+
+If you run into issues, try using `sudo`.
+
+Finally, run the image with:
+
+```shell
+docker run --rm inky-demo
+```
+
+And you should get a similar result as the previous time you run the script.
+
+## Step 4 (Optional): Composing Distroless Images in a Docker Multi-Stage Build
+
+As discussed in the introduction, it is possible to combine your fully-featured image with a distroless runtime in a Docker multistage build, and this will give you a final image that is also distroless. The [Getting Started with Python](/chainguard/chainguard-images/reference/python/getting-started-python/) tutorial shows in detail how to accomplish that using a `-dev` variant as **builder**, and the distroless Chainguard Python image as runtime. You can also accomplish the same results by using your newly built image based on `wolfi-base` in place of the `-dev` variant of the Python image. The only difference is that you'll need a few more steps to set up your builder image.
+
+The following Dockerfile uses a multi-stage build to obtain a final distroless image that contains everything the application needs to run. The build requires additional software that is not carried along to the final image.
+
+Open a new file and call it `DockerfileDistroless`:
+
+```shell
+nano DockerfileDistroless
+```
+
+Copy the following code into your new file:
+
+```
+FROM cgr.dev/chainguard/wolfi-base as builder
+
+RUN adduser -D nonroot
+WORKDIR /app
+
+RUN apk add python-3.11 py3.11-pip && \
+	chown -R nonroot.nonroot /app/
+
+USER nonroot
+COPY requirements.txt /app/
+RUN  pip3 install -r requirements.txt --user
+
+FROM cgr.dev/chainguard/python:latest
+
+WORKDIR /app
+
+# Make sure you update Python version in path
+COPY --from=builder /home/nonroot/.local/lib/python3.11/site-packages /home/nonroot/.local/lib/python3.11/site-packages
+
+COPY inky.py inky.png /app/
+
+ENTRYPOINT [ "python", "/app/inky.py" ]
+
+```
+Save and close the file when you're finished.
+
+
+Now, build this image using a custom tag so that you can compare the previously built `inky-demo` image with its distroless version:
+
+```shell
+docker build . -f DockerfileDistroless -t inky-demo:distroless
```

Comment:
Everything worked fine following Jason's Docker config suggestion, but at this build I'm receiving this error. Any ideas why? To me it seems like two arguments are being passed to the `COPY` statement...

```
[+] Building 0.1s (2/2) FINISHED                                                
 => [internal] load build definition from DockerfileDistroless             0.0s
 => => transferring dockerfile: 589B                                       0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
ERROR: failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to create LLB definition: dockerfile parse error line 18: COPY requires at least two arguments, but only one was provided. Destination could not be determined
```